### DISCUSSION TITLE
shield - documentation

### DISCUSSION THREAD

**Discussion by truefantomas** [codeigniter4/shield#234] [Ideas]
Hello friends!
The shield is a very interesting project that has been developing recently and is very necessary. We, those who create their own projects and do not delve deeply into the backend, really like the performance of this framework. But sometimes there are difficulties that we face. I would like you to think of us sometimes) and simplify the work so that we are focused on the frontend.
It would be great to implement a simpler connection to the main engine than it is now, at the on / off level.
Good documentation with implementation examples is also required, since the front-end shield covers very few target functions, their ability to describe is not difficult. Thanks

**Comment by kenjis** [comment]
> It would be great to implement a simpler connection to the main engine than it is now, at the on / off level.

I'm not sure *connection to the main engine* really means.

The installation steps would be the following two commands:
```
composer require codeigniter4/shield
php spark shield:setup
```
Do you mean this is not simple and you want a simpler way?

**Reply by lonnieezell** [reply]
I got an email from this person and they said they don't speak English and it's all Google Translate, but I believe they're talking about better docs, with more examples on usage so it's easier/faster and to get started with it.

**Reply by truefantomas** [reply]
you understood me correctly! the installation is complex, accompanied by many installations and corrections.  I completed it, and then I need to completely dive into the shield.  it's easy with codeigniter, I do small projects thanks to clear documentation that describes the functions of certain tasks.  so I wrote that it would be great to consider that developers of all levels of knowledge will understand your idea.

**Reply by truefantomas** [reply]
> > It would be great to implement a simpler connection to the main engine than it is now, at the on / off level.
> 
> I'm not sure _connection to the main engine_ really means.
> 
> The installation steps would be the following two commands:
> 
> ```
> composer require codeigniter4/shield
> php spark shield:setup
> ```
> 
> Do you mean this is not simple and you want a simpler way?

not quite easy!  in real life, the entry threshold for development is higher.  it took longer than expected to deal with all the accompanying installation problems.  I am the type of person who creates projects, small ones, it is quite difficult for me to quickly understand how the shield works.  use cases would solve this problem.

**Reply by kenjis** [reply]
> not quite easy! 

Where is difficult? 

I would like to improve the docs, but to be honest I don't know what is not easy and how to do it.
I don't get what are use cases you mean.

Or you can send a PR to improve the docs by yourself any time.
https://github.com/codeigniter4/shield/edit/develop/docs/install.md

**Comment by jadsy2107** [comment]
Ya, i second that - more documentation please and use cases. 

For instance how to list all users for an admin type control. Delete users, create users, edit users. 
I seem to have to look at Register.php and find how it creates the user and work back from there,

**Reply by MGatner** [reply]
This library does not include a CRUD interface for Users - that would add a whole new layer of opinion. You are encouraged to use the provided models to implement your own controllers for any of the resources but it is beyond the intended scope here.

**Comment by kenjis** [comment]
The documentation is now much improved.
https://github.com/codeigniter4/shield/blob/develop/docs/index.md
https://github.com/codeigniter4/shield/blob/develop/docs/quickstart.md
